File: typescript/resistor-color/resistor-color.ts
export const colorCode = (name: ResistorString) => {
  return COLORS.indexOf(name);
}

export const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white"
] as const;

type ResistorString = typeof COLORS[number];

File: elixir/resistor-color/lib/resistor_color.ex
defmodule ResistorColor do

  @resistor_values [
    :black,
    :brown,
    :red,
    :orange,
    :yellow,
    :green,
    :blue,
    :violet,
    :grey,
    :white
  ]

  @doc """
  Return the value of a color band
  """
  @spec code(
          :black
          | :blue
          | :brown
          | :green
          | :grey
          | :orange
          | :red
          | :violet
          | :white
          | :yellow
        ) :: non_neg_integer
  def code(code) when code in @resistor_values do
    Enum.find_index(@resistor_values, fn x -> x == code end)
  end

end
